Source/SpatialGDK/Private/SpatialActorChannel.cpp

```diff
@@ -557,7 +577,26 @@ void USpatialActorChannel::OnReserveEntityIdResponse(const worker::ReserveEntity
 	// If a Singleton was created, update the GSM with the proper Id.
 	if (Interop->IsSingletonClass(Actor->GetClass()))
 	{
-		Interop->UpdateGlobalStateManager(Actor->GetClass()->GetPathName(), ActorEntityId);
+		Interop->AddSingletonToGSM(Actor->GetClass()->GetPathName(), ActorEntityId);
+	}
+
+	if (Actor->IsFullNameStableForNetworking())
+	{
+		USpatialPackageMapClient* PackageMap = Cast<USpatialPackageMapClient>(SpatialNetDriver->GetSpatialOSNetConnection()->PackageMap);
+
+		uint32 CurrentOffset = 1;
```

Comment:
This code is a duplicate of some generated code, could be refactored.

Reply:
After discussion offline, this will be moved to OnRemoveComponent for the UnrealMetadata.